src/SmartTransactionsController.ts

```diff
@@ -1044,7 +1108,7 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
     const {
       smartTransactionsState: { smartTransactions },
     } = this.state;
-    const currentSmartTransactions = smartTransactions?.[this.#chainId];
+    const currentSmartTransactions = Object.values(smartTransactions).flat();
```

Comment:
nit: Same comment as I mentioned somewhere else, I would rename it to: `smartTransactionsForAllChains` and use that naming later on.